database/connector.py

```diff
@@ -1,207 +1,239 @@
+from typing import NoReturn, Optional, List, Tuple
+
 import sqlite3
 
-from datetime import datetime
+from database.new_types import (
+    User,
+    Event,
+    Expense,
+    Debt,
+)
+
 
 class Connector:
-    TABLES = [
-        'debts',
-        'user2event',
-        'expenses',
-        'users',
-        'events',
-    ]
-
-    def __init__(self):
+    def __init__(
+        self,
+        db_name: str = 'database.sqlite',
+    ):
         """
         Establishes connection to database etc.
         """
-        self.conn = sqlite3.connect('database.sqlite', check_same_thread=False)
+        self.conn = sqlite3.connect(
+            db_name,
+            check_same_thread=False,
+            isolation_level='EXCLUSIVE',
+        )
         self._create_database()
-        self.conn.commit()
 
     def _create_database(self) -> NoReturn:
         """
         Create database
         Maybe we need to manually apply the migration mechanism instead of this method
         """
         cursor = self.conn.cursor()
-        cursor.execute('PRAGMA foreign_keys=on;')
-        cursor.execute('''
-            CREATE TABLE IF NOT EXISTS users (
-                id   INTEGER PRIMARY KEY,
-                name VARCHAR
-            )
-        ''')
-        cursor.execute('''
-            CREATE TABLE IF NOT EXISTS events (
-                id    VARCHAR PRIMARY KEY,
-                name  VARCHAR NOT NULL UNIQUE
-            )
-        ''')
-        cursor.execute('''
-            CREATE TABLE IF NOT EXISTS user2event (
-                user_id INTEGER NOT NULL,
-                event_id INTEGER NOT NULL,
-                
-                FOREIGN KEY (user_id)  REFERENCES users(id),
-                FOREIGN KEY (event_id) REFERENCES events(id)
-            )
-        ''')
-        cursor.execute('''
-            CREATE TABLE IF NOT EXISTS expenses (
-                id        INTEGER PRIMARY KEY,
-                name      VARCHAR NOT NULL,
-                lender_id INTEGER NOT NULL,
-                event_id  INTEGER NOT NULL,
-                sum       INTEGER NOT NULL,
-                datetime  DATETIME NOT NULL,
-
-                FOREIGN KEY (lender_id) REFERENCES users(id),
-                FOREIGN KEY (event_id)  REFERENCES events(id)
-            )
-        ''')
-        cursor.execute('''
-            CREATE TABLE IF NOT EXISTS debts (
-                expense_id INTEGER NOT NULL,
-                lender_id  INTEGER NOT NULL,
-                debtor_id  INTEGER NOT NULL,
-                sum        FLOAT NOT NULL,
-
-                FOREIGN KEY (expense_id) REFERENCES expenses(id),
-                FOREIGN KEY (lender_id)  REFERENCES users(id),
-                FOREIGN KEY (debtor_id)  REFERENCES users(id)
-            )
-        ''')
+        current_dir = Path(__file__).resolve().parent
+        try:
+            with open(current_dir.joinpath('sql/create_tables.sql'), 'r') as file:
+                cursor.executescript(file.read())
```

Comment:
Replace this with [importlib-resources](https://importlib-resources.readthedocs.io/en/latest/using.html#example)

Reply:
Tech debt
Also we should not use importlib-resources, because I'm scared